### ISSUE TITLE
[GCM] How to send messages to a topic?

### ISSUE THREAD

**Issue by emaillenin** [rpush/rpush#219]
As mentioned here - https://developers.google.com/cloud-messaging/topic-messaging

**Comment by NeoElit** [comment]
Yeah.. I'm also looking forward to it. Any one working on this feature?

**Comment by NeoElit** [comment]
@emaillenin if you just want it to work, use this fork and provide all payload inside data field( including :data, :notification, :to ).
https://github.com/NeoElit/rpush/tree/implement_feature_gcm_topic_%23219

**Comment by aried3r** [comment]
Shouldn't this *just* work if you set `:to` to the topic?

**Comment by NeoElit** [comment]
@aried3r i didn't see any provision to set ```:to``` . So i just generalized these fields to existing ```:data``` and parsed each while dispatching for network request.

**Comment by zuk** [comment]
Would be great to see this implemented in mainline rpush.

**Comment by L3K0V** [comment]
Is there any up to date workaround or fix provided?

**Comment by aried3r** [comment]
I don't think anyone is working on this yet. But PRs are welcome, as always. I think using the legacy API it should be fairly simple to implement.

See _Topic HTTP POST request_ (way down in the legacy section) of https://firebase.google.com/docs/cloud-messaging/send-message.

This might also solve https://github.com/rpush/rpush/pull/293 since it also implements and requires a `to` key, just like topics.

However if I read the docs right, this doesn't enable you to send notifications to multiple topics.

**Comment by L3K0V** [comment]
Guys,

You can check this. I have implemented and currently I'm using in production this branch with support of FCM. Topics are supported and there are resolved some issues regarding FCM not accepting the payload. The changes are from July last year you can take a look and is kinda up-to date.

https://github.com/L3K0V/rpush/tree/fcm
